build/release/teamcity-publish-release.sh

```diff
@@ -1,4 +1,150 @@
 #!/usr/bin/env bash
 
-echo "Implement me!"
+set -euxo pipefail
 
+source "$(dirname "${0}")/teamcity-support.sh"
+
+
+if [[ -n "${PUBLISH_LATEST}" && -n "$PRE_RELEASE" ]]; then
+  echo "Invalid parameter combination: PUBLISH_LATEST and PRE_RELEASE can't both be set."
+  exit 6
+fi
+
+
+tc_start_block "Variable Setup"
+export BUILDER_HIDE_GOPATH_SRC=1
+
+# Matching the version name regex from within the cockroach code except
+# for the `metadata` part at the end because Docker tags don't support
+# `+` in the tag name.
+# https://github.com/cockroachdb/cockroach/blob/master/pkg/util/version/version.go#L75
```

Comment:
I'm worried about this link rotting - maybe using the permalink (https://github.com/cockroachdb/cockroach/blob/4c6864b44b9044874488cfedee3a31e6b23a6790/pkg/util/version/version.go#L75) would at least allow future readers to trace back the history if this changes?

Reply:
Indeed. Updated.